src/games/strategy/triplea/ai/proAI/ProNonCombatMoveAI.java

```diff
@@ -148,6 +146,8 @@ public ProNonCombatMoveAI(final ProAI ai) {
           break;
         }
       }
+    } else {
+      moveUnitsToBestTerritories();
```

Comment:
Yeah, this does not seem like the right fix, it's a copy/paste.
The while(true) on 116 looks like it has no effect given the break on 146. Seems like you could avoid the copy/paste by simply moving the existing call on line 133 to be outside of the if/else block.